### PR TITLE
VIH-11050 disconnect from camera and microphone when user leaves the waiting room

### PR DESCRIPTION
**Pull request by shaed-parkar**
### Jira link

VIH-11050

### HeartbeatService Improvements:
* Added an `initialising` flag to prevent multiple initializations of the heartbeat service. (`VideoWeb/VideoWeb/ClientApp/src/app/services/conference/heartbeat.service.ts`)

### UserMediaStreamService Enhancements:
* Introduced a `closeCurrentStream` method to stop all tracks of the current stream. (`VideoWeb/VideoWeb/ClientApp/src/app/services/user-media-stream.service.ts`)

### VideoCallService Updates:
* Updated the `disconnect` method to call the new `closeCurrentStream` method from `UserMediaStreamService`. (`VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/services/video-call.service.ts`)

### CODE REVIEW

File: VideoWeb/VideoWeb/ClientApp/src/app/services/user-media-stream.service.ts [hunk 1]
```diff
@@ -85,6 +85,12 @@ export class UserMediaStreamService {
         this.activeMicrophoneStreamSubject.next(stream);
     }
 
+    closeCurrentStream() {
```
Comment (oliver-scott) [inline VideoWeb/VideoWeb/ClientApp/src/app/services/user-media-stream.service.ts:88]: could we add a test for this?

**Review by oliver-scott — COMMENTED** [review]
looks good, just one suggestion

File: VideoWeb/VideoWeb/ClientApp/src/app/services/user-media-stream.service.ts (hunk 1 above)
Comment (shaed-parkar) [reply VideoWeb/VideoWeb/ClientApp/src/app/services/user-media-stream.service.ts:88]: we can, I've had some real double mocking MediaTracks and it just fails with spys. I;ll have another crack at it but I faced a similar issue in my other or with the v2 media stream service

Comment (oliver-scott) [reply VideoWeb/VideoWeb/ClientApp/src/app/services/user-media-stream.service.ts:88]: don't worry about it if it's going to be a pain, happy to approve for now so we can get it over the line and address later

**Review by oliver-scott — APPROVED** [review]

**Review by komaldabhi1509 — APPROVED** [review]

**Pull request merged**